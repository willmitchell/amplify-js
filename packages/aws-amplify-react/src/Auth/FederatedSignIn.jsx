import * as React from 'react';
import { Component } from 'react';

import { JS, I18n, ConsoleLogger as Logger } from '@aws-amplify/core';
import Auth from '@aws-amplify/auth';

import AmplifyTheme from '../Amplify-UI/Amplify-UI-Theme';
import {
    FormSection,
    SectionBody,
    Strike,
} from '../Amplify-UI/Amplify-UI-Components-React';

import {
    GoogleButton,
    FacebookButton,
    AmazonButton,
    OAuthButton
} from './Provider';

const logger = new Logger('FederatedSignIn');

export class FederatedButtons extends Component {
    google(google_client_id) {
        if (!google_client_id) { return null; }

        const { theme, onStateChange } = this.props;
        return <GoogleButton
                google_client_id={google_client_id}
                theme={theme}
                onStateChange={onStateChange}
              />
    }

    facebook(facebook_app_id) {
        if (!facebook_app_id) { return null; }

        const { theme, onStateChange } = this.props;
        return <FacebookButton
                facebook_app_id={facebook_app_id}
                theme={theme}
                onStateChange={onStateChange}
                />
    }

    amazon(amazon_client_id) {
        if (!amazon_client_id) { return null; }

        const { theme, onStateChange } = this.props;
        return <AmazonButton
                amazon_client_id={amazon_client_id}
                theme={theme}
                onStateChange={onStateChange}
              />
    }

    OAuth(oauth_config) {
        if (!oauth_config) { return null;}
        const { theme, onStateChange } = this.props;
        return <OAuthButton
                label={oauth_config? oauth_config.label : undefined}
                theme={theme}
                onStateChange={onStateChange}
              />
    }

    render() {
        const { authState } = this.props;
        if (!['signIn', 'signedOut', 'signedUp'].includes(authState)) { return null; }

        const federated = this.props.federated || {};
        if (!Auth || typeof Auth.configure !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }

        const config = Auth.configure();
        if (config.oauth) {
            federated.oauth_config = Object.assign({}, federated.oauth_config, config.oauth);
        }

        if (JS.isEmpty(federated)) { return null; }

        const { google_client_id, facebook_app_id, amazon_client_id, oauth_config } = federated;

        const theme = this.props.theme || AmplifyTheme;
        return (
            <div>
                <div>
                {this.google(google_client_id)}
                </div>
                <div>
                {this.facebook(facebook_app_id)}
                </div>
                <div>
                {this.amazon(amazon_client_id)}
                </div>
                <div>
                {this.OAuth(oauth_config)}
                </div>
                <Strike>{I18n.get('or')}</Strike>
            </div>
        )
    }
}

export default class FederatedSignIn extends Component {
    render() {
        const { authState, onStateChange } = this.props;
        let federated = this.props.federated || {};
        if (!Auth || typeof Auth.configure !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        
        const config = Auth.configure();
        if (config.oauth) {
            federated.oauth_config = Object.assign({}, federated.oauth_config, config.oauth);
        }

        if (!federated) {
            logger.debug('federated prop is empty. show nothing');
            logger.debug('federated={google_client_id: , facebook_app_id: , amazon_client_id}');
            return null;
        }
        if (!['signIn', 'signedOut', 'signedUp'].includes(authState)) { return null; }
        logger.debug('federated Config is', federated);
        const theme = this.props.theme || AmplifyTheme;
        return (
            <FormSection theme={theme}>
                <SectionBody theme={theme}>
                    <FederatedButtons
                        federated={federated}
                        theme={theme}
                        authState={authState}
                        onStateChange={onStateChange}
                    />
                </SectionBody>
            </FormSection>
        )
    }
}
