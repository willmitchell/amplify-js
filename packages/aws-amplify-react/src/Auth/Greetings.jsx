/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import * as React from 'react';
import { I18n, ConsoleLogger as Logger } from '@aws-amplify/core';
import Auth from '@aws-amplify/auth';
import AuthPiece from './AuthPiece';
import { NavBar, Nav, NavRight, NavItem, NavButton } from '../Amplify-UI/Amplify-UI-Components-React';

import AmplifyTheme from '../Amplify-UI/Amplify-UI-Theme';

const logger = new Logger('Greetings');

export default class Greetings extends AuthPiece {
    constructor(props) {
        super(props);

        this.signOut = this.signOut.bind(this);
        this.googleSignOut = this.googleSignOut.bind(this);
        this.facebookSignOut = this.facebookSignOut.bind(this);
       

        this.state = {
            authState: props.authState,
            authData: props.authData
        }
    }

    componentDidMount() {
        this._isMounted = true;
    }

    componentWillUnmount() {
        this._isMounted = false;
    }

    signOut() {
        this.googleSignOut();
        this.facebookSignOut();
        if (!Auth || typeof Auth.signOut !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        Auth.signOut()
            .then(() => this.changeState('signedOut'))
            .catch(err => { logger.error(err); this.error(err); });
    }

    googleSignOut() {
        const authInstance = window.gapi && window.gapi.auth2? window.gapi.auth2.getAuthInstance() : null;
        if (!authInstance) {
            return Promise.resolve(null);
        }

        authInstance.then((googleAuth) => {
            if (!googleAuth) {
                logger.debug('google Auth undefined');
                return Promise.resolve(null);
            }

            logger.debug('google signing out');
            return googleAuth.signOut();
        });
    }

    facebookSignOut() {
        const fb = window.FB;
        if (!fb) {
            logger.debug('FB sdk undefined');
            return Promise.resolve(null);
        }

        fb.getLoginStatus(response => {
            if (response.status === 'connected') {
                return new Promise((res, rej) => {
                    logger.debug('facebook signing out');
                    fb.logout(response => {
                        res(response);
                    });
                });
            } else {
                return Promise.resolve(null);
            }
        });
    }

    inGreeting(name) { return 'Hello ' + name; }
    outGreeting() { return ''; }

    userGreetings(theme) {
        const user = this.state.authData;
        const greeting = this.props.inGreeting || this.inGreeting;
        // get name from attributes first
        const nameFromAttr = user.attributes? 
            (user.attributes.name || 
            (user.attributes.given_name? 
                (user.attributes.given_name + ' ' + user.attributes.family_name) : undefined))
            : undefined;

        const name = nameFromAttr || user.name || user.username;
        const message = (typeof greeting === 'function')? greeting(name) : greeting;
        return (
            <span>
                <NavItem theme={theme}>{message}</NavItem>
                <NavButton
                    theme={theme}
                    onClick={this.signOut}
                >
                    {I18n.get('Sign Out')}
                </NavButton>

            </span>
        )
    }

    noUserGreetings(theme) {
        const greeting = this.props.outGreeting || this.outGreeting;
        const message = (typeof greeting === 'function')? greeting() : greeting;
        return message? <NavItem theme={theme}>{message}</NavItem> : null;
    }

    render() {
        const { hide } = this.props;
        if (hide && hide.includes(Greetings)) { return null; }

        const { authState } = this.state;
        const signedIn = (authState === 'signedIn');

        const theme = this.props.theme || AmplifyTheme;
        const greeting = signedIn? this.userGreetings(theme) : this.noUserGreetings(theme);
        if (!greeting) { return null; }

        return (
            <NavBar theme={theme}>
                <Nav theme={theme}>
                    <NavRight theme={theme}>
                        {greeting}
                    </NavRight>
                </Nav>
            </NavBar>
        )
    }
}
