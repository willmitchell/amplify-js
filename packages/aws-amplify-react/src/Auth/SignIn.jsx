/*
 * Copyright 2017-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import * as React from 'react';
import { I18n, JS, ConsoleLogger as Logger } from '@aws-amplify/core';
import Auth from '@aws-amplify/auth';

import AuthPiece from './AuthPiece';
import { FederatedButtons } from './FederatedSignIn';

import {
    FormSection,
    FormField,
    SectionHeader,
    SectionBody,
    SectionFooter,
    Button,
    Link,
    Hint,
    Input,
    InputLabel,
    SectionFooterPrimaryContent,
    SectionFooterSecondaryContent,
} from '../Amplify-UI/Amplify-UI-Components-React';

const logger = new Logger('SignIn');

export default class SignIn extends AuthPiece {
    constructor(props) {
        super(props);

        this.checkContact = this.checkContact.bind(this);
        this.signIn = this.signIn.bind(this);
        this.onKeyDown = this.onKeyDown.bind(this);

        this._validAuthStates = ['signIn', 'signedOut', 'signedUp'];
        this.state = {};
    }

    componentDidMount() {
        window.addEventListener('keydown', this.onKeyDown);
    }

    componentWillUnmount() {
        window.removeEventListener('keydown', this.onKeyDown);
    }

    onKeyDown(e) {
        if (this.props.authState === 'signIn' && !this.props.hide) {
            if (e.keyCode === 13) { // when press enter
                this.signIn();
            }
        }
    }

    checkContact(user) {
        if (!Auth || typeof Auth.verifiedContact !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        Auth.verifiedContact(user)
            .then(data => {
                if (!JS.isEmpty(data.verified)) {
                    this.changeState('signedIn', user);
                } else {
                    user = Object.assign(user, data);
                    this.changeState('verifyContact', user);
                }
            });
    }

    signIn() {
        const { username, password } = this.inputs;
        if (!Auth || typeof Auth.signIn !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        Auth.signIn(username, password)
            .then(user => {
                logger.debug(user);
                if (user.challengeName === 'SMS_MFA' || user.challengeName === 'SOFTWARE_TOKEN_MFA') {
                    logger.debug('confirm user with ' + user.challengeName);
                    this.changeState('confirmSignIn', user);
                } else if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
                    logger.debug('require new password', user.challengeParam);
                    this.changeState('requireNewPassword', user);
                } else if (user.challengeName === 'MFA_SETUP') {
                    logger.debug('TOTP setup', user.challengeParam);
                    this.changeState('TOTPSetup', user);
                }
                else {
                    this.checkContact(user);
                }
            })
            .catch(err => {
                if (err.code === 'UserNotConfirmedException') {
                    logger.debug('the user is not confirmed');
                    this.changeState('confirmSignUp');
                } 
                else if (err.code === 'PasswordResetRequiredException') {
                    logger.debug('the user requires a new password');
                    this.changeState('requireNewPassword');
                } else {
                    this.error(err);
                }
            });
    }

    showComponent(theme) {
        const { authState, hide = [], federated, onStateChange } = this.props;
        if (hide && hide.includes(SignIn)) { return null; }
        const hideSignUp = hide.some(component => component.name === 'SignUp')
        const hideForgotPassword = hide.some(component => component.name === 'ForgotPassword')
        return (
            <FormSection theme={theme}>
                <SectionHeader theme={theme}>{I18n.get('Sign in to your account')}</SectionHeader>
                <SectionBody theme={theme}>
                    <FederatedButtons
                        federated={federated}
                        theme={theme}
                        authState={authState}
                        onStateChange={onStateChange}
                    />
                    <FormField theme={theme}>
                        <InputLabel>{I18n.get('Username')} *</InputLabel>
                        <Input
                            autoFocus
                            placeholder={I18n.get('Enter your username')}
                            theme={theme}
                            key="username"
                            name="username"
                            onChange={this.handleInputChange}
                        />
                    </FormField>
                    <FormField theme={theme}>
                        <InputLabel>{I18n.get('Password')} *</InputLabel>
                        <Input
                            placeholder={I18n.get('Enter your password')}
                            theme={theme}
                            key="password"
                            type="password"
                            name="password"
                            onChange={this.handleInputChange}
                        />
                        {
                            !hideForgotPassword && <Hint theme={theme}>
                                {I18n.get('Forget your password? ')}
                                <Link theme={theme} onClick={() => this.changeState('forgotPassword')}>
                                    {I18n.get('Reset password')}
                                </Link>
                            </Hint>
                        }
                    </FormField>
                    
                </SectionBody>
                <SectionFooter theme={theme}>
                    <SectionFooterPrimaryContent theme={theme}>
                        <Button theme={theme} onClick={this.signIn}>
                            {I18n.get('Sign In')}
                        </Button>
                    </SectionFooterPrimaryContent>
                    {
                        !hideSignUp && <SectionFooterSecondaryContent theme={theme}>
                            {I18n.get('No account? ')}
                            <Link theme={theme} onClick={() => this.changeState('signUp')}>
                                {I18n.get('Create account')}
                            </Link>
                        </SectionFooterSecondaryContent>
                    }
                </SectionFooter>
            </FormSection>
        )
    }
}
