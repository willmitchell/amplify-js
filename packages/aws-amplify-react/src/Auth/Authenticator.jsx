/*
 * Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with
 * the License. A copy of the License is located at
 *
 *     http://aws.amazon.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
 * CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */

import * as React from 'react';
import { Component } from 'react';
import Amplify, { I18n, ConsoleLogger as Logger, Hub } from '@aws-amplify/core';
import Auth from '@aws-amplify/auth';
import Greetings from './Greetings';
import SignIn from './SignIn';
import ConfirmSignIn from './ConfirmSignIn';
import RequireNewPassword from './RequireNewPassword';
import SignUp from './SignUp';
import ConfirmSignUp from './ConfirmSignUp';
import VerifyContact from './VerifyContact';
import ForgotPassword from './ForgotPassword';
import TOTPSetup from './TOTPSetup';

import AmplifyTheme from '../Amplify-UI/Amplify-UI-Theme';
import AmplifyMessageMap from '../AmplifyMessageMap';

import { Container, Toast } from '../Amplify-UI/Amplify-UI-Components-React';

const logger = new Logger('Authenticator');

export default class Authenticator extends Component {
    constructor(props) {
        super(props);

        this.handleStateChange = this.handleStateChange.bind(this);
        this.handleAuthEvent = this.handleAuthEvent.bind(this);
        this.onHubCapsule = this.onHubCapsule.bind(this);

        this._initialAuthState = this.props.authState || 'signIn';
        this.state = { auth: 'loading' };
        Hub.listen('auth', this);
    }

    componentDidMount() {
        const config = this.props.amplifyConfig;
        if (config) {
            Amplify.configure(config);
        }
        this._isMounted = true;
        this.checkUser()
    }

    componentWillUnmount() {
        this._isMounted = false;
    }
    checkUser() {
        const { auth } = this.state;
        if (!Auth || typeof Auth.currentAuthenticatedUser !== 'function') {
            throw new Error('No Auth module found, please ensure @aws-amplify/auth is imported');
        }
        return Auth.currentAuthenticatedUser()
            .then(user => {
                if (!this._isMounted) { return; }
                if (auth !== 'signedIn') {
                    this.setState({
                        authState: 'signedIn',
                        authData: user
                    });
                    this.handleStateChange('signedIn', user);
                }
            })
            .catch(err => {
                if (!this._isMounted) { return; }
                Auth.signOut().then(() => this.handleStateChange(this._initialAuthState));
            });
    }

    onHubCapsule(capsule) {
        const { channel, payload, source } = capsule;
        if (channel === 'auth' && (payload.event === 'configured' || payload.event === 'cognitoHostedUI')) { 
            this.checkUser(); 
        }
    }

    handleStateChange(state, data) {
        logger.debug('authenticator state change ' + state, data);
        if (state === this.state.auth) { return; }

        if (state === 'signedOut') { state = 'signIn'; }
        this.setState({ auth: state, authData: data, error: null });
        if (this.props.onStateChange) { this.props.onStateChange(state, data); }
    }

    handleAuthEvent(state, event) {
        if (event.type === 'error') {
            const map = this.props.errorMessage || AmplifyMessageMap;
            const message = (typeof map === 'string')? map : map(event.data);
            this.setState({ error: message, showToast: true });
            
        }
    }

    render() {
        const { auth, authData } = this.state;
        const theme = this.props.theme || AmplifyTheme;
        const messageMap = this.props.errorMessage || AmplifyMessageMap;

        let { hideDefault, hide = [], federated } = this.props;
        if (hideDefault) {
            hide = hide.concat([
                Greetings,
                SignIn,
                ConfirmSignIn,
                RequireNewPassword,
                SignUp,
                ConfirmSignUp,
                VerifyContact,
                ForgotPassword,
                TOTPSetup
            ]);
        }
        const props_children = this.props.children || [];

        const default_children = [
            <Greetings/>,
            <SignIn federated={federated}/>,
            <ConfirmSignIn/>,
            <RequireNewPassword/>,
            <SignUp/>,
            <ConfirmSignUp/>,
            <VerifyContact/>,
            <ForgotPassword/>,
            <TOTPSetup/>
        ];

        const props_children_names  = React.Children.map(props_children, child => child.type.name)
        hide = hide.filter((component) =>!props_children_names.includes(component.name))
        const render_props_children = React.Children.map(props_children, (child, index) => {
            return React.cloneElement(child, {
                    key: 'aws-amplify-authenticator-props-children-' + index,
                    theme: theme,
                    messageMap: messageMap,
                    authState: auth,
                    authData: authData,
                    onStateChange: this.handleStateChange,
                    onAuthEvent: this.handleAuthEvent,
                    hide: hide
                });
        });
       
        const render_default_children = hideDefault ? [] : React.Children.map(default_children, (child, index) => {
                return React.cloneElement(child, {
                    key: 'aws-amplify-authenticator-default-children-' + index,
                    theme: theme,
                    messageMap: messageMap,
                    authState: auth,
                    authData: authData,
                    onStateChange: this.handleStateChange,
                    onAuthEvent: this.handleAuthEvent,
                    hide: hide
                });
            });

        const render_children = render_default_children.concat(render_props_children);
        const error = this.state.error;        

        return (
            <Container theme={theme}>
                {this.state.showToast && 
                    <Toast onClose={() => this.setState({showToast: false})}>
                        { I18n.get(error) }
                    </Toast>
                }
                {render_children}
            </Container>
        )
    }
}
