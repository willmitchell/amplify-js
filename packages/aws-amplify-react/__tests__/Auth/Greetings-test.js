import Auth from '@aws-amplify/auth';
import Greetings from '../../src/Auth/Greetings';
import * as React from 'react';
import AmplifyTheme from '../../src/AmplifyTheme';
import AuthPiece from '../../src/Auth/AuthPiece';
import { Header, Footer, Input, Button } from '../../src/Amplify-UI/Amplify-UI-Components-React';

const acceptedStates = [
    'signedIn'
];

const deniedStates = [
    'signIn',  
    'signedUp', 
    'signedOut',
    'forgotPassword',
    'signUp',
    'confirmSignIn',
    'confirmSignUp',
    'verifyContact'
];

describe('Greetings', () => {
    describe('normal case', () => {
        test('render correctly with authState signedIn', () => {
            const wrapper = shallow(<Greetings/>);
            for (var i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    theme: 'theme'
                });
                expect(wrapper).toMatchSnapshot();
            }
        });

        test('render correctly with hide', () => {
            const wrapper = shallow(<Greetings/>);
            for (var i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    theme: 'theme',
                    hide: [Greetings]
                });
                expect(wrapper).toMatchSnapshot();
            }
        });

        test('render name from attributes', () => {
            const wrapper = shallow(<Greetings/>);
            wrapper.setProps({
                authState: 'signedIn',
                theme: 'theme'
            });

            wrapper.setState({
                authData: {
                    attributes: {
                        name: 'name'
                    }
                },
                authState: 'signedIn'
            })  

            expect(wrapper).toMatchSnapshot();
        })
    });

   
    test('render corrently with other authStates', () => {
        const wrapper = shallow(<Greetings/>);
        
        for (var i = 0; i < deniedStates.length; i += 1){
            wrapper.setProps({
                authState: deniedStates[i],
                theme: 'theme'
            });

            expect(wrapper).toMatchSnapshot();
        }
    });

    describe('signOut test', () => {
        test('happy case', async () => {
            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();

            const spyon = jest.spyOn(Auth, 'signOut').mockImplementationOnce(() => {
                return Promise.resolve();
            });

            await greetings.signOut();

            expect(spyon).toBeCalled();
            spyon.mockClear();
        });

        test('error case', async () => {
            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();

            const spyon = jest.spyOn(Auth, 'signOut').mockImplementationOnce(() => {
                return Promise.reject('error');
            });

            await greetings.signOut();

            expect(spyon).toBeCalled();
            spyon.mockClear();
        });
    });

    describe('google signOut test', () => {
        test('happy case', async () => {
            const mockFn = jest.fn();

            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        return Promise.resolve({
                            signOut: mockFn
                        })
                    }
                }
            };

            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();

            await greetings.googleSignOut();

            expect(mockFn).toBeCalled();
        });

        test('no auth2', async () => {
            window.gapi = null;
            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();

            expect(await greetings.googleSignOut()).toBeNull();
        });

        test('no googleAuth', async () => {
            window.gapi = {
                auth2: {
                    getAuthInstance() {
                        return Promise.resolve(null);
                    }
                }
            };

            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();

            await greetings.googleSignOut();
        });
    });

    describe('facebook signout test', () => {
        test('happy case', async () => {
            window.FB = {
                getLoginStatus(callback) {
                    callback({
                        status: 'connected'
                    })
                },
                logout(callback) {
                    callback('response');
                }
            }

            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();
            
            await greetings.facebookSignOut()
        });

        test('not connected', async () => {
            window.FB = {
                getLoginStatus(callback) {
                    callback({
                        status: 'not connected'
                    })
                },
                logout(callback) {
                    callback('response');
                }
            }
            const wrapper = shallow(<Greetings/>);
            const greetings = wrapper.instance();
            
            await greetings.facebookSignOut()
        });
    });
});
