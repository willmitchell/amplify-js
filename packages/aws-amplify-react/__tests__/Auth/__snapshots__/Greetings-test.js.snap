// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greetings normal case render correctly with authState signedIn 1`] = `""`;

exports[`Greetings normal case render correctly with hide 1`] = `""`;

exports[`Greetings normal case render name from attributes 1`] = `
<Component
  theme="theme"
>
  <Component
    theme="theme"
  >
    <Component
      theme="theme"
    >
      <span>
        <Component
          theme="theme"
        >
          Hello name
        </Component>
        <Component
          onClick={[Function]}
          theme="theme"
        >
          Sign Out
        </Component>
      </span>
    </Component>
  </Component>
</Component>
`;

exports[`Greetings render corrently with other authStates 1`] = `""`;

exports[`Greetings render corrently with other authStates 2`] = `""`;

exports[`Greetings render corrently with other authStates 3`] = `""`;

exports[`Greetings render corrently with other authStates 4`] = `""`;

exports[`Greetings render corrently with other authStates 5`] = `""`;

exports[`Greetings render corrently with other authStates 6`] = `""`;

exports[`Greetings render corrently with other authStates 7`] = `""`;

exports[`Greetings render corrently with other authStates 8`] = `""`;
